33515 invert remove foreign key support "to_table"

jychen7 opened the pull request:
### Summary

remove_foreign_key supports
- remove_foreign_key :accounts, :branches
- remove_foreign_key :accounts, to_table: :branches

but the second one is not reversible.

This branch is to fix and allow second one to be reversible.

### Other Information

close https://github.com/rails/rails/issues/33515
continuing https://github.com/rails/rails/pull/33519, base on discussion between @nicktime and @sikachu

nicktime commented:
Pls, add info on :to_table option, it's missing, something like
```
diff --git a/activerecord/lib/active_record/connection_adapters/abstract/schema_statements.rb b/activerecord/lib/active_record/connection_adapters/abstract/schema_statements.rb
index 9d46eb5..7e54340 100644
--- a/activerecord/lib/active_record/connection_adapters/abstract/schema_statements.rb
+++ b/activerecord/lib/active_record/connection_adapters/abstract/schema_statements.rb
@@ -988,7 +988,10 @@ def add_foreign_key(from_table, to_table, options = {})
       #
       #   remove_foreign_key :accounts, name: :special_fk_name
       #
-      # The +options+ hash accepts the same keys as SchemaStatements#add_foreign_key.
+      # The +options+ hash accepts the same keys as SchemaStatements#add_foreign_key
+      # with an addition of
+      # [<tt>:to_table</tt>]
+      # +to_table+ contains the referenced primary key.
       def remove_foreign_key(from_table, options_or_to_table = {})
         return unless supports_foreign_keys?
 ```

jychen7 commented:
### CI

[All GREEN except only one job](https://travis-ci.org/rails/rails/builds/412259070)

```
No output has been received in the last 10m0s, this potentially indicates a stalled build or something wrong with the build itself.
Check the details on how to adjust your build configuration on: https://docs.travis-ci.com/user/common-build-problems/#Build-times-out-because-no-output-was-received
The build has been terminated
```

How to rerun?

schneems commented:
I restarted the failing job

schneems commented:
Everything looks good. Can you squash to one commit. You can give credit to multiple authors using this technique https://contributors.rubyonrails.org/faq

jychen7 commented:
@schneems I just squash and force push.

kamipo commented:
The real author name of #33519 is *Nikolay Epifanov*.

https://patch-diff.githubusercontent.com/raw/rails/rails/pull/33519.patch

jychen7 commented:
@kamipo commit message is updated to use real author name

jychen7 committed:
Allow `to_table` in `invert_remove_foreign_key`

remove_foreign_key supports
- remove_foreign_key :accounts, :branches
- remove_foreign_key :accounts, to_table: :branches

but the second one is not reversible.

This branch is to fix and allow second one to be reversible.

[Nikolay Epifanov, Rich Chen]

jychen7 commented:
@schneems changelog is added.

ps: looks like changelog is easier to conflic

jychen7 commented:
@schneems @kamipo sorry, will merge soon? need any update from me?

jychen7 commented:
CI was green. No code changes in this branch.
I close the PR by accident, then re-open it, the CI failed at `i18n`? 

```
Failure:
ApplicationTests::I18nTest#test_config.i18n.fallbacks.map_=_{_:ca_=>_:'es-ES'_}_initializes_fallbacks_with_a_mapping_ca_=>_es-ES [test/application/initializers/i18n_test.rb:240]:
expected fallbacks for :ca to be [:ca, :"es-ES", :es, :en], but were [:ca, :"es-ES", :es].
Expected: [:ca, :"es-ES", :es, :en]
  Actual: [:ca, :"es-ES", :es]
rails test test/application/initializers/i18n_test.rb:237
```

schneems commented:
I restarted the tests

jychen7 commented:
@schneems thanks, but I think restart doesn't work, need https://github.com/rails/rails/pull/33566

schneems commented:
Since that's merged can you rebase and re-push?

jychen7 commented:
@schneems thanks, I rebase and force push.

only one job fail, need restart this time

jychen7 commented:
@schneems @kamipo CI green now, ready to merge?

schneems commented:
Thanks!

jychen7 commented:
@schneems thanks
